client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -118,23 +121,27 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		const planSlug = getUpgradePlanSlugFromPath( this.props.product, this.props.selectedSite );
-
-		let cartItem,
-			cartMeta;
-
-		if ( planSlug ) {
-			cartItem = getCartItemForPlan( planSlug );
+		const { product, productsList, purchaseId, selectedSite, selectedSiteSlug } = this.props;
+		const meta = product.split( ':' )[ 1 ];
+		const productSlug = getUpgradePlanSlugFromPath( product, selectedSite.jetpack ) || product.split( ':' )[ 0 ];
```

Comment:
Any chance to have the cart already loaded from the server, but sites to not be there yet? I know it sounds like an edge case, but just to be sure, can we use the `isJetpackSite` selector instead of raw `.jetpack`?